### ISSUE TITLE
[feature request] Create Kubernetes objects from the extension

### ISSUE THREAD

**Issue by radu-matei** [Azure/vscode-kubernetes-tools#347]
I would like to be able to create Kubernetes objects from the extension. The experience would be as follows:

- right click the type of object you want to create from the extension pane
- click create
- the editor opens YAML file, pre-populated with the template for the object type I want to create

Of course it would not work for all objects in the tree (such as Nodes), but for anything that can be created with a declarative file, it should work.

Thoughts?

**Comment by squillace** [comment]
What's the user story for this, as a developer? Can you write the scenario out? Radu might want that, or Ralph, but for prioritization I'd love to hear the "common developer workflow" story.

**Comment by radu-matei** [comment]
User story:

- Radu is working on a simple application. Suddenly, he needs to add credentials for a database connection string, and he heard that secrets might be the way to go in Kubernetes. As he never created a secret before, or is unfamiliar / unwilling to write Kubernetes YAML by hand, from scratch, this would spare Radu the Google search for Kubernetes secret, then the copy-paste from the Kubernetes docs, a right-click -> create secret -> pre-populated YAML file might be the experience he needs.

Now replace <secret> with any Kubernetes object. Does it make sense?

Thanks!

**Comment by squillace** [comment]
yes, more generally, I would start with specific objects and work up from there. Secrets/configmaps are good starts. :-)

**Comment by vdefeo** [comment]
Hey guys, would be also nice to edit ConfigMaps and Secrets from the extension. :)

**Comment by itowlson** [comment]
Raised a new issue #579 for the specific case of ConfigMaps and Secrets because those are more specific, we have more concrete use cases and it's a bit more bounded.
